tools/socktap/link_layer.cpp

```diff
@@ -29,7 +30,43 @@ create_link_layer(boost::asio::io_service& io_service, const EthernetDevice& dev
         namespace ip = boost::asio::ip;
         ip::udp::endpoint multicast(ip::address::from_string("239.118.122.97"), 8947);
         link_layer.reset(new UdpLink { io_service, multicast });
+    } else if (name.substr(0, 3) == "tcp") {
+        namespace ip = boost::asio::ip;
+
+        TcpLink* tcp = new TcpLink { io_service };
+
+        std::string tcp_ip;
+        unsigned short tcp_port, tcp_ip_len;
+
+        if (vm.count("tcp-connect")) {
+            for (const std::string& ip_port : vm["tcp-connect"].as<std::vector<std::string>>()) {
+                tcp_ip_len = ip_port.find(":");
+                tcp_ip = ip_port.substr(0, tcp_ip_len);
+                tcp_port = std::stoi(ip_port.substr(tcp_ip_len + 1));                
+                tcp->connect({ip::address::from_string(tcp_ip), tcp_port});
+            }
+        }
+
+        if (vm.count("tcp-accept")) {
+            for (const std::string& ip_port : vm["tcp-accept"].as<std::vector<std::string>>()) {
+                tcp_ip_len = ip_port.find(":");
```

Comment:
I would like to extract the string parsing of IP address and port to a separate function. In this function, we could also do proper error handling. For example, is there a default port if the colon ":" is missing?